Whisky/Views/Updater/UpdatePreviewView.swift

```diff
@@ -0,0 +1,169 @@
+                     : String(format: String(localized: "update.descriptionLoaded"),
+                              "v" + currentVersion,
+                              nextVersion))
+                Spacer()
+                HStack {
+                    Button("update.cancel") {
+                        dismiss()
+                    }
+                    Spacer()
+                    Button("update.update") {
+                        Task(priority: .userInitiated) {
+                            install()
+                        }
+                    }
+                    .buttonStyle(.borderedProminent)
+                    .keyboardShortcut(.defaultAction)
+                }
+            }
+            .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .leading)
+            .padding(20)
+            VStack {
+                if markdownTextState == .loading {
+                    ProgressView()
+                } else if markdownTextState == .error {
+                    VStack(spacing: 12) {
+                        Text("update.changeLogFailed")
+                        Button("update.retryChangeLog") {
+                            Task(priority: .userInitiated) {
+                                await getChangelog()
+                            }
+                        }
+                    }
+                } else {
+                    ScrollView {
+                        VStack(alignment: .leading) {
+                            Text("update.changeLog")
+                                .font(.title2)
+                                .fontWeight(.bold)
+                                .padding(.bottom, 12)
+                        }
+                        .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .leading)
+                        Markdown {
+                            markdownText
+                        }
+                        .markdownTheme(.basic)
+                    }
+                    .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .leading)
+                }
+            }
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
+            .padding(20)
+            .background(.ultraThickMaterial)
+        }
+        .frame(maxWidth: .infinity, maxHeight: .infinity)
+        .onAppear {
+            Task(priority: .userInitiated) {
+                await getChangelog()
+            }
+        }
+    }
+
+    func getChangelog() async {
+        withAnimation { markdownTextState = .loading }
+        let ghOwner = (Bundle.main.object(forInfoDictionaryKey: "GithubRepoOwner") as? String) ?? "Whisky-App"
+        let ghRepo = (Bundle.main.object(forInfoDictionaryKey: "GithubRepoName") as? String) ?? "Whisky"
+
+        // Make a request to the Github API to get the latest release
+        // Append path not using string interpolation to non-urlencoded paths
+        guard let url = URL(string: "https://api.github.com/")?
```

Comment:
Can this not use the current appcast? We can't connect to GitHub as it's blocked in China.

Reply:
Okay, it's now using the xml file as requested